include/ray.h

```diff
@@ -9,20 +9,28 @@ class Ray {
 private:
   Vertex origin_;
   Direction direction_;
+  float importance_;
```

Comment:
That's why we need an  `int iteration_count;` that's passed and updated recursively and checked against a `const int MAX_ITERATION_DEPTH;` ☕️